main.c

```diff
@@ -694,12 +695,60 @@ _install_transaction (Pool         *pool,
     {
       Id p = installedq.elements[x];
       queue_pushunique (pile, p);
-      g_debug ("%*c - %s", indent, ' ', pool_solvid2str (pool, p));
+      const gchar *solvable = pool_solvid2str (pool, p);
+      g_debug ("%*c - %s", indent, ' ', solvable);
+      /* Non-modules are immediately resolved as resolved, since for RPM the
+       * result would not change if done again. However for modules we need to
+       * make sure we look at all combinations. */
+      if (!g_str_has_prefix (solvable, "module:"))
+        map_set (tested, p);
     }
 
   return TRUE;
 }
 
+/**
+ * mask_bare_rpms:
+ * @pool: initialized pool
+ *
+ * For each available modular package, find all bare RPMs with the same name,
+ * and mark them as not considered.
+ */
+static void
+mask_bare_rpms(Pool *pool)
+{
+  Id *available_modular_packages = pool_whatprovides_ptr (pool, pool_str2id (pool, MODPKG_PROV, 1));
+  for (Id *pp = available_modular_packages; *pp; pp++)
+    {
+      Solvable *modpkg = pool_id2solvable (pool, *pp);
+      const gchar *name = pool_id2str (pool, modpkg->name);
+
+      g_auto(Queue) sel;
+      queue_init (&sel);
+      selection_make (pool, &sel, name, SELECTION_NAME);
+      if (!sel.count)
+        {
+          /* This should never happen, at least one package
+           * should match (the modular one). */
+          continue;
+        }
+      g_auto(Queue) q;
+      queue_init (&q);
+      selection_solvables (pool, &sel, &q);
+
+      for (int j = 0; j < q.count; j++)
```

Comment:
I wonder if you could replace this for loop with `selection_subtract` to remove the modular packages from the selection.

Reply:
agree